packages/ic-management/README.md

```diff
@@ -232,7 +233,23 @@ Creates a canister. Only available on development instances.
 | ------------------------------------- | ------------------------------------------------------------------------------------------------------- |
 | `provisionalCreateCanisterWithCycles` | `({ settings, amount, canisterId, }?: ProvisionalCreateCanisterWithCyclesParams) => Promise<Principal>` |
 
-[:link: Source](https://github.com/dfinity/ic-js/tree/main/packages/ic-management/src/ic-management.canister.ts#L296)
+[:link: Source](https://github.com/dfinity/ic-js/tree/main/packages/ic-management/src/ic-management.canister.ts#L300)
+
+##### :gear: ecdsaPublicKey
+
+Calculate a SEC1 encoded ECDSA public key for the given canister using the given derivation path. If the `canister_id` is unspecified, it will default to the canister id of the caller. The `derivation_path` is a vector of variable length byte strings. Each byte string may be of arbitrary length, including empty. The total number of byte strings in the `derivation_path` must be at most 255. The `key_id` is a struct specifying both a `curve` and a `name`. The availability of a particular `key_id` depends on implementation.
+
+| Method           | Type                                                                                                |
+| ---------------- | --------------------------------------------------------------------------------------------------- |
+| `ecdsaPublicKey` | `({ keyId, canisterId, derivationPath, }: EcdsaPublicKeyParams) => Promise<EcdsaPublicKeyResponse>` |
+
+Parameters:
+
+- `params.keyId`: The key id for which the public key will be derived, consisting of a `name` and a `curve`.
+- `params.canisterId`: The canister id for which the public key will be derived. It defaults to the caller's canister id if unspecified.
+- `params.derivationPath`: The derivation path that will be used to derive the public key.
```

Comment:
do we know now how to get that `derivationPath` on the frontend side? if yes can we document it?

Reply:
So, the derivation path, when talking about BTC address, is a sequence of strings and numbers that follows the "rules" defined in [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki).
```
m / purpose' / coin_type' / account' / change / address_index
```
For example, a basic derivation path for Bitcoin Testnet would be `m/44'/1'/0'/0/0`. Ultimately, as a generic function, it will be the user's choice.

Do you want to create some checks on it for the frontend? Format checks for example? I will add the link to BIP-44 there, in the meanwhile.

Reply:
Thanks for the feedback. Not for this PR but, does it accept any generic derivation path or are the paths limited to those of [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) ? If the latest, indeed we should probably be more descriptive in what is accepted, that will help the developers (consumers) of the library. WDYT?

PS: Again not for this PR.

Reply:
Theoretically, it can accept any path (in fact, even empty), because it is basically a mathematical formula. But the standard/rule is to accept BIP-44.

So, I agree with you that we should guide the users to help "build" the derivation path that they would need.

Reply:
Created PR #654 about this